app/scripts/modules/core/src/presentation/tables/TableContext.tsx

```diff
@@ -0,0 +1,14 @@
+import * as React from 'react';
+
+import { ITableLayout, ITableColumn } from './Table';
+
+export interface ITableContext {
+  layout: ITableLayout;
+  isMobile: boolean;
+  expandable: boolean;
+  allExpanded: boolean;
+  setAllExpanded: (allExpanded: boolean) => any;
+  columns: ITableColumn[];
+}
+
+export const TableContext = React.createContext<ITableContext>(null);
```

Comment:
While this context-based approach is _slightly_ roundabout when passing things around, I prefer it over using render props because basically **every table related component** in the subtree of a `<Table/>` needs a slice of this context in some way or another, and you'd end up with `{...props}` in a gazillion places without getting much of any noticeable benefit (at least as far as I can tell). Happy to talk about other approaches if folks have suggestions, but would prefer to avoid too much error-prone manual wiring.